enhancements/testing/openshift-tests-extension.md

```diff
@@ -0,0 +1,642 @@
+    # overdraw on allocatable pod resources for a set of
+    # tests run in parallel.
+    "memory": "1Gi",
+    
+    # If a duration can be roughly predicted, inclusion of this
+    # information may improve the test execution 
+    # planning algorithm performance.  
+    "duration": "2s",
+      
+    # Timeout information may also support efficient
+    # planning.
+    "timeout": "16s",
+  },
+
+  # Tests can be identified as "informing" or "production".
+  # Informing tests will not negatively impact default views
+  # in Component Readiness.
+  # However, policies may be established in the future that 
+  # cause tests that remain informing too long to be treated
+  # as if they are production.
+  "lifecycle": "informing",
+  
+}
+```
+
+##### Run-Test - Running Extension Tests
+The `run-test` verb is used to cause the extension binary to actually run 
+discovered tests. `run-test` accepts test environment arguments as discussed
+in the "list" verb section (in order to reduce fact discovery time needed
+for each test run) and one or more test names to invoke.
+```
+$ extension-binary run-test 
+  --platform      The hardware or cloud platform ("aws", "gcp", "metal", ...).
+  ...other context arguments...
+  --config        The component configuration profile to assume is active on the cluster.
+  --name | -n     Test name to invoke (-n can be specified multiple times).
+```
+
+`run-test` should run the specified tests in parallel. To prevent 
+test cases developing a dependence on one another, `run-test` must make no
+assumption about the number or composition of tests it is asked to run
+(other than they will be consistent with the test environment and isolation
+requirements defined by the "list" verb).
+
+`openshift-tests` will randomize tests into multiple different parallel
+executions of the extension binary in order to amortize initialization
+costs.
+
+Standard output from the invocation will be serialized into JSONL with
+each line associated with a single test run result.
+
+```python
+{ "name": "T1", "result": "success", }
+{ "name": "T1", "result": "success", }
+{ "name": "T1", "result": "success", }
+```
+
+A single test result will contain the following information, encoded on a single line:
+```python
+{
+    "name": "test name",
+    
+    # Duration of test run in milliseconds
+    "duration": "1012",
+    
+    # The outcome of the test; pass, fail, skip, timeout.
+    "result": "pass",
+    "output": "standard output of the test",
+    "error": "standard error of the test",
+    
+    # Human-readable messages to further explain 
+    # skips / timeouts / etc.
+    "messages": [
+        "example message"
+    ]
+}
+```
+
+##### Config - Component Configuration Testing
+
+A component can advertise that it wants to be exercised in multiple different configuration.
+`openshift-tests` will plan an efficient method of testing those configurations and
+ensure tests appropriate to that configuration are run.
+

```

Comment:
What happens when you run the same tests against multiple configurations? How do we report those results?

I think we need to describe how that works. A simple way to handle this would be make sure the results reported in the JUnit append `[Config:XYZ]` to the name.

Reply:
`Test Result Aggregation` touches on this and suggests we move beyond junit. A single test result is a rich structure including information about success/failure AND the configuration in which it was run.